src/cargo/core/compiler/build_config.rs

```diff
@@ -56,15 +56,15 @@ impl BuildConfig {
     /// * `target.$target.libfoo.metadata`
     pub fn new(
         config: &Config,
-        jobs: Option<u32>,
+        jobs: Option<i32>,
         keep_going: bool,
         requested_targets: &[String],
         mode: CompileMode,
     ) -> CargoResult<BuildConfig> {
         let cfg = config.build_config()?;
         let requested_kinds = CompileKind::from_requested_targets(config, requested_targets)?;
         if jobs == Some(0) {
-            anyhow::bail!("jobs must be at least 1")
+            anyhow::bail!("jobs may not be 0")
         }
```

Comment:
This test seems to be duplicated below. Would it be possible to remove this?

Reply:
Removed